### ISSUE TITLE
PackedByteArray: add to_vector2_array and to_vector3_array functions

### ISSUE THREAD

**Issue by wareya** [godotengine/godot-proposals#8139]
### Describe the project you are working on

A custom node with a custom editor that needs to compress diffed packed vector3 arrays in the undo buffer to conserve memory.

### Describe the problem or limitation you are having in your project

[PackedVector3Array](https://docs.godotengine.org/en/stable/classes/class_packedvector3array.html#class-packedvector3array) has a [to_byte_array](https://docs.godotengine.org/en/stable/classes/class_packedvector3array.html#class-packedvector3array-method-to-byte-array) function, but [PackedByteArray](https://docs.godotengine.org/en/stable/classes/class_packedbytearray.html#class-packedbytearray) doesn't have a function to do the reverse.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add the relevant missing functions to PackedByteArray.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

PackedByteArray would have new functions.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

Doing this from script code is slow.

### Is there a reason why this should be core and not an add-on in the asset library?

Enhancement to built-in type.

**Comment by wareya** [comment]
Functions converting to PackedStringArray and PackedColorArray are missing too.

**Comment by KoBeWi** [comment]
I think you can use `decode_var()` for that.

**Comment by AThousandShips** [comment]
Code to convert to `PackedStringArray` wouldn't work at the moment as encoding to that doesn't work itself

**Comment by wareya** [comment]
> I think you can use decode_var() for that.

Maybe. `decode_var` doesn't allow you to specify the type, so it can only be used in conjunction with `encode_var`, and `encode_var` requires sufficient space to be allocated in the byte array ahead of time, but it's not documented in the description of `encode_var` how much extra space is needed for type information.

**Comment by AThousandShips** [comment]
There's already a PR implementing this btw:
* https://github.com/godotengine/godot/pull/76075

**Comment by wareya** [comment]
Looking through the docs more, `var_to_bytes` and `bytes_to_var` work for me, so I'm not blocked by this.

**Comment by blackears** [comment]
I'd like to add to_color_array() would also be an important addition.

**Comment by MrBeeMovie** [comment]
Any idea when this is going to be merged into engine? It is necessary for a project that I am working on that uses compute shaders. It is way too slow to manually decode bytes from buffer using gdscript.

**Comment by nvl-ez** [comment]
> Any idea when this is going to be merged into engine? It is necessary for a project that I am working on that uses compute shaders. It is way too slow to manually decode bytes from buffer using gdscript.

Same here...
